### ISSUE TITLE
Duplication warnings on different documents

### ISSUE THREAD

**Issue by kskyten** [papis/papis#30]
For some reason I'm getting duplication warnings when I try to add two different documents.
```bash
(papis) ➜  papis mkdir lib
(papis) ➜  papis papis -l lib add --from-url https://arxiv.org/abs/1712.02621v1
(papis) ➜  papis papis -l lib add --from-url https://arxiv.org/abs/1711.09534v1

WARNING:Command:                    
abstract:   Generating novel, yet realistic, images of persons is a challenging task due
to the complex interplay between the different image factors, such as the
foreground, background and pose information. In this work, we aim at generating
such images based on a novel, two-stage reconstruction pipeline that learns a
disentangled representation of the aforementioned image factors and generates
novel person images at the same time. First, a multi-branched reconstruction
network is proposed to disentangle and encode the three factors into embedding
features, which are then combined to re-compose the input image itself. Second, three corresponding mapping functions are learned in an adversarial manner in
order to map Gaussian noise to the learned embedding feature space, for each
factor respectively. Using the proposed framework, we can manipulate the
foreground, background and pose of the input image, and also sample new
embedding features to generate such targeted manipulations, that provide more
control over the generation process. Experiments on Market-1501 and Deepfashion
datasets show that our model does not only generate realistic person images
with new foregrounds, backgrounds and poses, but also manipulates the generated
factors and interpolates the in-between states. Another set of experiments on
Market-1501 shows that our model can also be beneficial for the person
re-identification task.
archiveprefix:   arXiv
author:   Liqian Ma and Qianru Sun and Stamatios Georgoulis and Luc Van Gool and Bernt Schiele and Mario Fritz
eprint:   1712.02621v1
file:   1712.02621v1.pdf
files:   ['tmpbm24n56y']
month:   Dec
primaryclass:   cs.CV
ref:   1712.02621v1
title:   Disentangled Person Image Generation
type:   article
url:   http://arxiv.org/abs/1712.02621v1
year:   2017

WARNING:Command:DUPLICATION WARNING
WARNING:Command:The document above seems to be already in your libray: 

WARNING:Command:(Hint) Use the update command if you just want to update the info.
Really add? (Y/n): y
```

**Comment by alejandrogallo** [comment]
Funny, I don't seem to be able to reproduce it,
[video](https://asciinema.org/a/rTmGwu8F0wseU5iNlVWd39IR5).

Two possible reasons come to mind, although I'm not entirely sure.
 - Do you by any chance have a
library called `lib` too?
 - What happens if you clear the cache, i.e., 
```
papis --cc add --from-url ......
```
Thanks for the report !

**Comment by kskyten** [comment]
I shoudn't have a library called lib. Here's my config:
```
[papers]
dir = ~/papers

[settings]
default-library = papers
```
I get the same error when I run:
```bash
rm -rf lib
mkdir lib
papis --cc -l lib add --from-url https://arxiv.org/abs/1712.02621v1
papis -l lib add --from-url https://arxiv.org/abs/1711.09534v1
```
Where is the cache stored?

**Comment by alejandrogallo** [comment]
weird!

The cache is stored in
```
~/.cache/papis
```
also depending on your XDG cache home, but if you did not change the
default variable, it should be there.

**Comment by nkoep** [comment]
I run into the same issue. I start with a clean install of papis, create a single library called `papers`, and call `papis add --from-url` with two different URLs. The first call goes through fine and the document ends up in the library. On the second call with a different URL papis complains with a duplication warning about the first document. I installed papis via pip in miniconda, but I ran into the same problem when I installed it with the system-wide pip.

I already tried disabling the cache in the config, as well as cleaning the cache with the same result.

**Comment by Xunius** [comment]
I got same problem.
I think the duplication check is too greedy, currently (I'm looking at v0.6) it looks at `doi`, `ref`, `isbn`, isbn10` and `url` in order, and if any of those matches it reports a duplicate.

My documents end up having same `isbn` numbers, and those are fetched by the Mendeley software. I'm not 100 % sure about how `isbn` and `doi` identifiers work, but all my duplicates are from the same journal (Journal of Climate), and they all have the same isbn (08948755), so I think it's not a good unique identifier.

**Comment by alejandrogallo** [comment]
Hi @Xunius ,

According to [wikipedia's isbn entry](https://en.wikipedia.org/wiki/International_Standard_Book_Number)

```
An ISBN is assigned to each edition and variation (except reprintings) of a book. For example, an e-book, a paperback and a hardcover edition of the same book would each have a different ISBN. 
```

so apparently Mendeley might be doing something nasty there... There is no reason why different
books, even different editions of the same book would have the same isbn.

This also applies to the doi, therefore I do not think the isbn is a bad identifier in itself...
However is good to review these aspects again, thanks so much for your engagement!

**Comment by Xunius** [comment]
Hi, alejandrogallo

I've found that wiki, and some others, but still not sure whether 2 articles can share the same isbn. Can I interpret that definition as that a journal is a "book" so multiple articles in that "book" all have the same isbn? In fact if I download the bib citation from the journal there isn't a isbn field at all, Mendeley is fetching is extra piece and causing trouble.
Another question: if 2 duplicates are different by one having full author names, and the other using initials for first-mid names, I guess they won't detected as duplicates?

**Comment by alejandrogallo** [comment]
Hi @Xunius,

it's interesting, I guess in some cases the `isbn` will not be good, so I thought about
making it user-configurable

**Comment by alejandrogallo** [comment]
Solved, it was a bug in the docmatcher, it will be solved in v0.8
